lib/amazon-pricing/helpers/instance-type.rb

```diff
@@ -165,16 +168,26 @@ def family_members(api_name)
         all_instances.select { |family, instances| instances.include?(api_name) }.values.first
       end
 
-
       def api_name_to_nf(name)
         type = name.split('.').last
         if (type == METAL)
+          # See if our metal instance has a hard-coded nf value
+          if !metal_to_nf[name].nil?
+            return metal_to_nf[name]
+          end
           # try to get largest size supported for family: presumes METAL is *not* in size_to_nf hash
           # assumes family_members are sorted by size
           sizes = family_members(name)
+          # Return nil if we have a bogus instance type
+          if sizes.nil?
+            return nil
+          end
           type = sizes[-1].split('.').last        # 'metal' defaults to largest size
           if sizes[-1].split('.').last == METAL
-            type = sizes[-2].split('.').last      # 'metal' already largest, so 2nd largest      
+            if sizes.size == 1 # We have an instance family with only metals, return 1 as the NF
+              return 1
```

Comment:
this means we failed to find this `name` in `metal_to_nf`, and, potentially this NF is totally bogus.  I think we should decide what an unknown NF should be for this case, if we're trying never to be wrong?

Reply:
Does it matter what the NF is if there is only one size (metal) in a family? I don't think so.

I would argue this case is not likely to happen, as we are the ones maintaining instance definitions, and would likely be abreast of any new instance families.

Reply:
@jeffzhou791 do you have any problem with us using 1 as a 'default' NF for metal instances where there is only 'metal' in a family? Assuming we try and apply the correct NF value using the `metal_to_nf` value first?

Reply:
a default value here feels wrong - think we should raise